Tests/SwiftDocCTests/Model/SemaToRenderNodeMultiLanguageTests.swift

```diff
@@ -888,6 +888,57 @@ class SemaToRenderNodeMixedLanguageTests: XCTestCase {
         )
     }
     
+    func testDoesNotFilterArticlesBasedOnSymbolAndArticleLanguageVariant() throws {
+        let outputConsumer = try renderNodeConsumer(
+            for: "MixedLanguageFramework",
+            configureBundle: { bundleURL in
+                try """
+                # ObjCArticle
+
+                @Metadata {
+                    @SupportedLanguage(objc)
+                }
+
+                This article has Objective-C as the source language.
+                """.write(to: bundleURL.appendingPathComponent("ObjCArticle.md"), atomically: true, encoding: .utf8)
+                try """
+                # ``MixedLanguageFramework``
+                
+                This symbol has a Swift and Objective-C variant.
+
+                ## Topics
+
+                - <doc:ObjCArticle>
+                - ``_MixedLanguageFrameworkVersionNumber``
+                """.write(to: bundleURL.appendingPathComponent("MixedLanguageFramework.md"), atomically: true, encoding: .utf8)
+            }
+        )
+        assertIsAvailableInLanguages(
+            try outputConsumer.renderNode(
+                withTitle: "ObjCArticle"
+            ),
+            languages: ["occ"],
+            defaultLanguage: .objectiveC
+        )
+        assertIsAvailableInLanguages(
+            try outputConsumer.renderNode(
+                withTitle: "_MixedLanguageFrameworkVersionNumber"
+            ),
+            languages: ["occ"],
+            defaultLanguage: .objectiveC
+        )
+        let mixedLanguageFrameworkRenderNode = try outputConsumer.renderNode(withTitle: "MixedLanguageFramework")
+        
+        // Verify that the curated ObjC articles don't get dropped from the Swift variant of a symbol extension file.
+        XCTAssertTrue(
+            mixedLanguageFrameworkRenderNode.topicSections.flatMap(\.identifiers).contains(["doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/ObjCArticle"])
+        )
+        // Verify that the curated ObjC symbols do get dropped from the Swift variant of a symbol extension file.
+        XCTAssertFalse(
+            mixedLanguageFrameworkRenderNode.topicSections.flatMap(\.identifiers).contains(["doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/_MixedLanguageFrameworkVersionNumber"])
+        )
```

Comment:
Since this is testing behavior of language specific content it would be good to verify all language variants here. For example  
```suggestion
        // Language specific articles are included in all variants topic sections.
        XCTAssertTrue(swiftTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/ObjCArticle"))
        XCTAssertTrue(objcTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/ObjCArticle"))
        
        XCTAssertTrue(swiftTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/SwiftArticle"))
        XCTAssertTrue(objcTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/SwiftArticle"))
        // Language specific symbols are dropped in topic section variants for languages where the symbol isn't available.
        XCTAssertFalse(swiftTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/_MixedLanguageFrameworkVersionNumber"))
        XCTAssertTrue(swiftTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/SwiftOnlyClass"))
        
        XCTAssertTrue(objcTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/_MixedLanguageFrameworkVersionNumber"))
        XCTAssertFalse(objcTopicIDs.contains("doc://org.swift.MixedLanguageFramework/documentation/MixedLanguageFramework/SwiftOnlyClass"))
```

Reply:
Similar test assertions could also be applied to the other test.

Reply:
It took me a while to figure out how to obtain the values of `objcTopicIDs` since manually accessing the variant traits and patch was a bit confusing. I ended up following the code of the tests from `RenderNodeTranslatorSymbolVariantsTests` and ended up with the following code:


```swift
let data = try renderNode.encodeToJSON()
let variantRenderNode = try RenderNodeVariantOverridesApplier()
    .applyVariantOverrides(in: data, for: [.interfaceLanguage("occ")])
let objCRenderNode = try RenderJSONDecoder.makeDecoder().decode(RenderNode.self, from: variantRenderNode)
// Topic identifiers in the ObjC variant of the `MixedLanguageFramework` symbol
let objCTopicIDs = objCRenderNode.topicSections.flatMap(\.identifiers)
```

I couldn't find a way to access the "occ" topic variants without encoding the render node, so perhaps there's another (and better) way to achieve this.